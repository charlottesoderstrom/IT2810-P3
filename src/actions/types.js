export const FETCH_MOVIE = 'FETCH_MOVIE';
export const RESET_MOVIE = 'RESET_MOVIE';
export const EXPAND_MOVIE = "EXPAND_MOVIE";
export const UPDATE_TEXT = 'UPDATE_TEXT';
export const EXPAND = 'EXPAND';
export const FILTERSORT_CHANGED = 'FILTERSORT_CHANGED';
export const TAB_CHANGED = 'TAB_CHANGED';
export const LOAD_MORE = 'LOAD_MORE';
export const WORD_CLICKED = 'WORD_CLICKED';
export const RESET_PAGE = 'RESET_PAGE'; 
export const RESET_TEXT = 'RESET_TEXT';  
export const RESET_FILTERSORT = 'RESET_FILTERSORT';  





