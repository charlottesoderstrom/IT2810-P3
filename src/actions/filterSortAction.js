import {FILTERSORT_CHANGED} from './types';
import {RESET_FILTERSORT} from './types';


// Gets triggered when someone checked/unchecked a filter category or sort by date
export const filterSortChange = (category, checked) => {
    return {
        type: FILTERSORT_CHANGED,
        category,
        checked,
    }
}

// Gets triggered when the user changed tab from the search page
export const resetFilterSort = () => {
    return {
        type: RESET_FILTERSORT
    }
}