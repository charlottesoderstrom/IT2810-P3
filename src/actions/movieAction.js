import {FETCH_MOVIE} from './types'; 
import {RESET_MOVIE} from './types'; 

/* Fetch movies only by the title, max 10. Gets page so the backend know which movies to give
Checks if sort is checked as well*/
export const fetchMoviesByTitle = (title, page, byDate) => dispatch =>{
    let sort;
    if(!byDate) {
        sort = "alpha"
    }
    else if(byDate) {
        sort = "date"
    }
    let search = title.replace(" ", "+")
    fetch("http://localhost:3000/"+sort+"/title="+search+"&page="+page)
    .then(res => res.json())
    .then(resJSON => 
        dispatch({
            type: FETCH_MOVIE,
            payload: resJSON,
        })
    );
};

/* When there is an empty search, all movies will be fetched, 10 for each 
time this function is called. Gets page so the backend know which movies to give
   Checks if sort is checked as well*/
export const fetchAllMovies = (page, byDate) => dispatch =>{
    let sort;
    if(!byDate) {
        sort = "alpha"
    }
    else if(byDate) {
        sort = "date"
    }
    fetch("http://localhost:3000/"+sort+"/all&page=" + page)
    .then(res => res.json())
    .then(resJSON => 
        dispatch({
            type: FETCH_MOVIE,
            payload: resJSON,
        })
    );
};

/* Fetches max 10 movies with no search input, but with filtered categories
Gets page so the backend know which movies to give
   Checks if sort is checked as well */
export const fetchFilteredMovies = (page, genres, byDate) => dispatch =>{
    let sort;
    if(!byDate) {
        sort = "alpha"
    }
    else if(byDate) {
        sort = "date"
    }
    var genreList = ""
    for (var i = 0; i < genres.length; i++) {
        genreList += genres[i] + "+";
      }
    genreList = genreList.slice(0,-1)
    fetch("http://localhost:3000/"+sort+"/genres="+genreList+"&page=" + page)
    .then(res => res.json())
    .then(resJSON => 
        dispatch({
            type: FETCH_MOVIE,
            payload: resJSON,
        })
    );
};

/* Fetch max. 10 movies with search input, filtered categories. 
Gets page so the backend know which movies to give
   Checks if sort is checked as well */
export const fetchFilteredTitleMovies = (page, title, genres, byDate) => dispatch =>{
    let sort;
    if(!byDate) {
        sort = "alpha"
    }
    else if(byDate) {
        sort = "date"
    }

    var genreList = ""
    for (var i = 0; i < genres.length; i++) {
        genreList += genres[i] + "+";
      }
    genreList = genreList.slice(0,-1)
    let search = title.replace(" ", "+")

    fetch("http://localhost:3000/f/"+sort+"/title="+search+"&genre="+genreList+"&page=" + page)
    .then(res => res.json())
    .then(resJSON => 
        dispatch({
            type: FETCH_MOVIE,
            payload: resJSON,
        })
    );
};

/* Gets triggered when the user clicks on a category in the wordcloud
Fetches 10 movies in a given category*/
export const fetchOneCategory = (category, page) => dispatch =>{
    fetch("http://localhost:3000/alpha/genres="+category+"&page=" + page)
    .then(res => {
        return res.json()
    })
    .then(resJSON => {
        dispatch({
            type: FETCH_MOVIE,
            payload: resJSON,
        })
    }
    );
};

/* Gets triggered when the user click on "Most popular tab"
Fetches 10 movies, sorted by number of votes, showing the highest
number of votes first*/
export const fetchMostPopular = (page) => dispatch =>{
    fetch("http://localhost:3000/vote/all&page=" + page)
    .then(res => {
        return res.json()
    })
    .then(resJSON => {
        dispatch({
            type: FETCH_MOVIE,
            payload: resJSON,
        })
    }
    );
};

/* Resets the movie result saved in the store. Gets triggered when the user
changes tab or is done with a search, i.e clicked on the search button*/
export const handleReset = () =>{
    console.log("movie reset reached")
    return {
        type: RESET_MOVIE
    }
};


