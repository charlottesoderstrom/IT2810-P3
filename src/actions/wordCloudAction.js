import {WORD_CLICKED} from './types';

// Gets triggered when the user has clicked on a category in the wordcloud
export const handleWordClicked = (category) => {
    return {
        type: WORD_CLICKED,
        category
    }
}



