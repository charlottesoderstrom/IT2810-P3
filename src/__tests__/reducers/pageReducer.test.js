import React from 'react'; 
import pageReducer from '../../reducers/pageReducer'; 
import TestRunner from 'jest-runner';
import {LOAD_MORE} from '../../actions/types';
import {RESET_PAGE} from '../../actions/types';

describe('page reducer', () => {
    it('should return the initial state', () => {
        expect(pageReducer(1,{type: LOAD_MORE})).toEqual(2);
    })
})

describe('page reducer', () => {
    it('should return the initial state', () => {
        expect(pageReducer(1,{type: RESET_PAGE})).toMatchSnapshot();
    })
})