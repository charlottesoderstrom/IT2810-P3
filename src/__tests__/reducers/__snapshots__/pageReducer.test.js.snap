// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page reducer should return the initial state 1`] = `0`;
